src/service/RecordsChunk.ts

```diff
@@ -16,8 +16,13 @@ export class RecordsChunk {
    * Get the next chunk of data based on the current chunk
    */
   public async getNextChunk(): Promise<RecordsChunk | null> {
-    const { index, limit, status } = this.meta ?? {}
-    const response = await this.session.api.getRecordsByStatus(this.session, status, index, limit)
+    const { index, limit, status, totalRecords } = this.meta ?? {}
+    let { skip } = this.meta ?? {}
+    if (skip >= totalRecords) {
+      // totalRecords is actually total remaining records
+      skip = 0
+    }
+    const response = await this.session.api.getRecordsByStatus(this.session, status, skip, limit)
```

Comment:
I think what we actually want here is `skip = 0`, generally I don't see a need for skip at all

Reply:
Each `chunk` fetches rows from the API using the `getFinalDatabaseView` endpoint with a status of `submitted` and returns that to the `onData` event. 

On the `RecordChunkIterator`, the `afterEach` method is called, changing the status of this chunk of rows to either `review` for partial rejections, or `accepted` otherwise - now when the next batch fetches rows, the `totalRecords` returned by the `getFinalDatabaseView` are only accounting for the remaining rows with a `submitted` status. 

Now that I walked through it, I see that the skip is very necessary on the `getNextChunk` method